src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -185,29 +187,29 @@ public int GetCountOfAllCheepFromAuthor(string author)
     public async Task<Cheep> Create(CheepCreateDTO cheep)
     {
         //NullReferenceException is handled in the constructor - CheepRepository()
-        var validationResult = await _validator.ValidateAsync(cheep);
+        ValidationResult result = _validator.Validate(cheep);
 
-        if (!validationResult.IsValid)
+        if (!result.IsValid)
         {
             throw new ValidationException();
         }
+        
+        var user = db.Authors.SingleAsync(u => u.Name == cheep.Author);
 
-        var user = await db.Authors.SingleAsync(u => u.Name == cheep.Author);
-
-        var entity = new Cheep{
-            Text = cheep.Text,
-            Author = user,
-            TimeStamp = DateTime.Now
-        };
-
-        db.Add(new Cheep
+        try
+        {
+             db.Add(new Cheep
                 {
-                    Author = entity.Author,
-                    Text = entity.Text,
-                    TimeStamp = entity.TimeStamp
+                    Author = user.Result,
+                    Text = cheep.Text,
+                    TimeStamp = DateTime.Now
                 });
-        await db.SaveChangesAsync();
-
-        return entity;
+        }
+        catch (System.Exception)
+        {
+            throw;
+        }
```

Comment:
Why do we have the try-catch block here when it just throws the exception again, and does nothing else?
Couldn't we just except it, since it already throws the exception?

Reply:
The code states this:

        if (!result.IsValid)
        {
            throw new ValidationException();
        }
        
        var user = db.Authors.SingleAsync(u => u.Name == cheep.Author);

        try
        {
             db.Add(new Cheep
                {
                    Author = user.Result,
                    Text = cheep.Text,
                    TimeStamp = DateTime.Now
                });
        }
        catch (System.Exception)
        {
            throw;
        }
        
So we're not throwing the same exception twice, this is two different exeptions, handling two different things

Reply:
I notice that. But I am talking purely about the try catch block, not the validationexception.

I have resolved this in commit 3115815